internal/pkg/provision/providers/docker/result.go

```diff
@@ -5,13 +5,23 @@
 package docker
 
 import (
+	"fmt"
+
 	"github.com/talos-systems/talos/internal/pkg/provision"
 )
 
 type result struct {
 	clusterInfo provision.ClusterInfo
 }
 
+func (res *result) Provisioner() string {
+	return "docker"
+}
+
 func (res *result) Info() provision.ClusterInfo {
 	return res.clusterInfo
 }
+
+func (res *result) StateDirectoryPath() (string, error) {
```

Comment:
Just pointing this out to support the argument for changing `--cluster-dir` to `--state`. Aligns with the interface, which is nice in this case.

Reply:
this is a bit different, this is path to `~/.talos/clusters/<name>` (state for specific cluster), but I think overall moving to `StateDirectory` instead of `ClustersDirectory` sounds good to me